app/controllers/gift_requests_controller.rb

```diff
@@ -48,6 +49,17 @@ def catalog
     @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
   end
 
+  def reserve
+    @gift_request = GiftRequest.find(params[:id])
+    reservation = Reservation.new(gift_request_id: params[:id], donor_id: current_user.id)
+    if reservation.save
+      redirect_to wishlist_path(@gift_request.list.id)
+    else
+      render "gift_request"
```

Comment:
This'll show the gift request editing form, which I don't think is what we want. I think redirecting to `catalog_path` either way and putting a relevant flash message depending on its success is the best way to go.